### PR TITLE
refactor: trigger pipeline with finalized hash

### PR DESCRIPTION
**Pull request by mattsse**
this renames the `restore_tree_if_possible` function (confusing name) and changes how the pipeline is potentially triggered again: using the finalized hash of the current FCU

### CODE REVIEW

File: crates/consensus/beacon/src/engine/mod.rs [hunk 1]
```diff
@@ -1165,8 +1164,18 @@ where
...
-                                Ok(_) => self.sync_state_updater.update_sync_state(SyncState::Idle),
+                            match self.update_tree_on_finished_pipeline(
+                                sync_target_state.finalized_block_hash,
+                            ) {
+                                Ok(synced) => {
+                                    if !synced {
+                                        // // We don't have the finalized block in the database, so
+                                        // we need to run another pipeline.
+                                        self.sync.set_pipeline_sync_target(
+                                            sync_target_state.finalized_block_hash,
```
Comment (rkrasiuk) [inline crates/consensus/beacon/src/engine/mod.rs:1175]: what's the motivation behind this change? we cannot add blocks on top of finalized unless they are in the block buffer. that potentially means more pipeline runs before live sync

**Review by rkrasiuk — CHANGES_REQUESTED** [review]

File: crates/consensus/beacon/src/engine/mod.rs (hunk 1 above)
Comment (Rjected) [reply crates/consensus/beacon/src/engine/mod.rs:1175]: Running the pipeline with the finalized hash seems safer than with the head at least, I don't really see an issue with this

**Review by Rjected — APPROVED** [review]
this seems fine, we might want to outline what failure modes we have w.r.t. the pipeline and node restarts in the future, which should inform which pipeline sync targets are safe and which ones are not

File: crates/consensus/beacon/src/engine/mod.rs [hunk 2]
```diff
@@ -1165,8 +1164,18 @@ where
...
                         {
                             // Update the state and hashes of the blockchain tree if possible.
-                            match self.restore_tree_if_possible(sync_target_state) {
-                                Ok(_) => self.sync_state_updater.update_sync_state(SyncState::Idle),
+                            match self.update_tree_on_finished_pipeline(
+                                sync_target_state.finalized_block_hash,
+                            ) {
+                                Ok(synced) => {
+                                    if !synced {
+                                        // // We don't have the finalized block in the database, so
```
Comment (onbjerg) [inline crates/consensus/beacon/src/engine/mod.rs:1172]:
```suggestion
                                        // We don't have the finalized block in the database, so
```

nit

**Review by onbjerg — APPROVED** [review]

**Review by rkrasiuk — APPROVED** [review]
